tests/test_updater_validation.py

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env python
+
+# Copyright 2022, New York University and the TUF contributors
+# SPDX-License-Identifier: MIT OR Apache-2.0
+
+"""Test ngclient Updater validations.
+"""
+
+import os
+import sys
+import tempfile
+import unittest
+
+import securesystemslib.exceptions
+
+from tests import utils
+from tests.repository_simulator import RepositorySimulator
+from tuf import unittest_toolbox
+from tuf.api.metadata import TargetFile
+from tuf.exceptions import FetcherHTTPError
+from tuf.ngclient import Updater
+
+
+class TestUpdater(unittest_toolbox.Modified_TestCase):
+    """Test ngclient Updater input validation."""
+
+    def setUp(self) -> None:
+        # pylint: disable-next=consider-using-with
+        self.temp_dir = tempfile.TemporaryDirectory()
+        self.metadata_dir = os.path.join(self.temp_dir.name, "metadata")
+        self.targets_dir = os.path.join(self.temp_dir.name, "targets")
+        os.mkdir(self.metadata_dir)
+        os.mkdir(self.targets_dir)
+
+        # Setup the repository, bootstrap client root.json
+        self.sim = RepositorySimulator()
+        with open(os.path.join(self.metadata_dir, "root.json"), "bw") as f:
+            f.write(self.sim.signed_roots[0])
+
+    def tearDown(self) -> None:
+        self.temp_dir.cleanup()
+
+    def _new_updater_with_targets_setup(self) -> Updater:
+        return Updater(
+            self.metadata_dir,
+            "https://example.com/metadata/",
+            self.targets_dir,
+            "https://example.com/targets/",
+            fetcher=self.sim,
+        )
+
+    def _new_updater_without_targets_setup(self) -> Updater:
+        return Updater(
+            self.metadata_dir,
+            "https://example.com/metadata/",
+            fetcher=self.sim,
+        )
+
+    def _new_updater_with_non_existing_metadata_dir(self) -> Updater:
+        return Updater(
+            "non_existing_metadata_dir",
+            "https://example.com/metadata/",
+            fetcher=self.sim,
+        )
+
+    def test_empty_target_path_validation(self) -> None:
+        updater = self._new_updater_with_targets_setup()
+        target_file = updater.get_targetinfo("")
+        self.assertIsNone(target_file)
+
+    def test_empty_targetinfo_validation(self) -> None:
+        updater = self._new_updater_without_targets_setup()
+        info = TargetFile(1, {"sha256": ""}, "")
+        with self.assertRaisesRegex(
+            ValueError, "target_dir must be set if filepath is not given"
+        ):
+            updater.download_target(info)
+
+        updater = self._new_updater_with_targets_setup()
+        with self.assertRaisesRegex(FetcherHTTPError, "No target"):
```

Comment:
Even if we decide to use `assertRaisesRegex` in order to additionally make sure the exception is the right one I am not sure we need it now.
`FetcherHTTPError` is our custom exception and we don't expect it to come from elsewhere.

Reply:
I can agree here if most arguments are against using regex. On the other hand `FetcherHTTPError` can be thrown for a variety of reasons, but I'm fine checking the exception only. Let's see what others say.